src/forms/CalendarComponent.jsx

```diff
@@ -3,151 +3,90 @@ import React from 'react';
-	componentWillReceiveProps(newProps) {
-		if (this.flatpickr) {
-			if (newProps.datepickerOptions) {
-				this.updateFlatpickrOptions(newProps.datepickerOptions);
-			}
-			this.flatpickr.setDate(newProps.value);
-		}
+		this.onFlatPickerChange = this.onFlatPickerChange.bind(this);
 	}
 
 	/**
-	 * Updates config options on the flatpickr instance
-	 * @param {Object} newOptions new configaration options for flatpickr
-	 * @return {undefined}
+	 * @function onFlatPickerChange
+	 * @param {Array} selectedDates - list of recently selected dates from flatpickr
+	 *
+	 * @description the Flatpickr component always passes an array of recently selected
+	 * dates to its onChange handler, with the most recent in first position of the array.
+	 * `redux-form` however, expects a single value. This function ensures that any `onChange`
+	 * prop passed to this component invokes with a single date object.
 	 */
-	updateFlatpickrOptions(newOptions) {
-		Object.keys(newOptions)
-			.filter(
-				option => this.props.datepickerOptions[option] !== newOptions[option]
-			)
-			.forEach(option => {
-				this.flatpickr.set(option, newOptions[option]);
-			});
-	}
-
-	/**
-	* @function onFlatpickrChange
-	* @param Array selectedDates
-	* @param dateStr
-	* @param Object instance the calendar instance
-	* @description signature conforms to the onChange handler flatpickr expects
-	* calls onChange if prop provided (eg from redux-form or wrapping components like DateTimePicker)
-	*/
-	onFlatpickrChange(selectedDates, dateStr, instance) {
+	onFlatPickerChange(selectedDates) {
 		this.props.onChange && this.props.onChange(selectedDates[0]);
 	}
 
-	/**
-	* @function onOpen
-	* @description event hook for flatpickr, used to call onFocus
-	* and apply focus highlight if this is a DateTimePicker
-	*/
-	onOpen() {
-		this.props.onFocus && this.props.onFocus();
-	}
-
-	/**
-	* @function onClose
-	* @description event hook for flatpickr, used to call onBlur
-	* and remove focus highlight if this is a DateTimePicker
-	*/
-	onClose() {
-		this.props.onBlur && this.props.onBlur();
-	}
-
 	render() {
 		const {
-			className,
+			id,
 			name,
 			label,
-			required,
-			value,
+			helperText,
 			error,
-			datepickerOptions, // eslint-disable-line no-unused-vars
+			required,
+			datepickerOptions,
+			className,
 			onChange, // eslint-disable-line no-unused-vars
-			helperText,
 			...other
 		} = this.props;
 
 		const classNames = {
 			label: cx(
 				{
 					required,
-					'flush--bottom': helperText
-				},
-				className
+					'flush--bottom': helperText,
+				}
 			),
 			helperText: cx(
 				'helperTextContainer',
 				{ required }
 			),
 			field: cx(
 				'input--dateTimePicker select--reset',
-				{ 'field--error': Boolean(error) },
+				{
+					'field--error': Boolean(error)
+				},
 				className
 			)
 		};
 
+		const options = {
+			altInput: true,
+			altFormat: 'D M d, Y',
+			nextArrow: `<span class="svg svg--chevron-right">
```

Comment:
Do we have to use markup? Can't we just return it from the `<Icon/>` component?

Reply:
I believe we do; these options get passed directly to `flatpickr` and we need to pass an html string

Reply:
Ok, I figured it was something like that